src/ring_theory/power_series/basic.lean

```diff
@@ -980,6 +980,28 @@ lemma is_unit_constant_coeff (φ : power_series R) (h : is_unit φ) :
   is_unit (constant_coeff R φ) :=
 mv_power_series.is_unit_constant_coeff φ h
 
+/-- Split off the constant coefficient. -/
+lemma eq_shift_mul_X_add_const (φ : power_series R) :
+  φ = mk (λ p, coeff R (p + 1) φ) * X + C R (constant_coeff R φ) :=
+begin
+  ext (_ | n),
+  { simp only [ring_hom.map_add, constant_coeff_C, constant_coeff_X, coeff_zero_eq_constant_coeff,
+    zero_add, mul_zero, ring_hom.map_mul], },
+  simp only [coeff_succ_mul_X, coeff_mk, linear_map.map_add],
+  simp only [coeff_C, n.succ_ne_zero, sub_zero, if_false, add_zero],
```

Comment:
This is just one goal, right? Does it work if you combine the two `simp only`s?
```suggestion
  { simp only [coeff_succ_mul_X, coeff_mk, linear_map.map_add,
      coeff_C, n.succ_ne_zero, sub_zero, if_false, add_zero],
```
(and if so, can you do the same in the other lemma?

Reply:
That works, cool.